### PR TITLE
Clean up headers in rp2_common to appease -Wstrict-prototypes

### PR DESCRIPTION
**Pull request by swetland**
In C, func() is a function taking an unspecified number of arguments,
vs func(void) a function taking no arguments. In C++ both forms indicate
"no arguments."
    
Update these headers to use the (void) form, which is correct in both
languages and avoids complaints when -Wstrict-prototypes is specified.

### CODE REVIEW

**Review by kilograham — COMMENTED** [review]
Can you add `-Wstrict-prototypes` to`test/kitchen_sink/CMakeLists.txt`and fixup the remaining issues from building that (it builds by default if you build from the SDK)
(Note worth doing both a release and debug build since some stuff is #ifdef-ed)

**Comment by swetland** [comment]
Sure. Almost done, but it turns out that:
cc1plus: error: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++ [-Werror]

Do you know the Cmake incantation to add it just for C sources?

Also, how do you specify a debug vs release build?

**Comment by kilograham** [comment]
you asked :-)

`$<$<COMPILE_LANGUAGE:CXX>:-Wstrict-prototypes>`

For debug, if not using an IDE which probably does this for you, you can pass `-DCMAKE_BUILD_TYPE=xxx` for e.g. Debug or Release, or the others..

Best to keep separate build directories for each config 

you can look at 

`cmake -LH` to see some more info

**Comment by swetland** [comment]
Updated patch. Looks clean for me in Debug and Release build modes.

**Comment by lurch** [comment]
image

File: test/kitchen_sink/CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,7 @@ target_compile_options(kitchen_sink_options INTERFACE
         -Wfloat-equal
         -Wmissing-format-attribute
         -Wno-long-long
+	$<$<COMPILE_LANGUAGE:C>:-Wstrict-prototypes>
```
Comment (kilograham) [inline test/kitchen_sink/CMakeLists.txt:62]: can you please fix the indent (8 spaces)

**Review by kilograham — CHANGES_REQUESTED** [review]
otherwise great, thank you

**Review by kilograham — APPROVED** [review]
lgtm

File: test/kitchen_sink/CMakeLists.txt (hunk 1 above)
Comment (swetland) [reply test/kitchen_sink/CMakeLists.txt]: Done.  Also re-based against origin/develop and fixed new instance of the warning in unqiue_iq.c

Pushed revised branch.

**Pull request merged**